### PR TITLE
feat: add GitHub Action out-of-the-box

### PR DESCRIPTION
**Pull request by vinayakkulkarni**
- [x] Enables GitHub actions CI workflow out of the box
	- [x] If user selects one of the test runners, `yarn|npm run test` will be added to `.github/workflows/ci.yml`
	- [x] If user selects one of the linting options, `yarn|npm run lint` will be added to `.github/workflows/ci.yml`
- [x] If `github-actions` option is selected from CI (while scaffolding new nuxt app), `dependabot` config is auto-updated to support the `github-actions` ecosystem
- [x] Debug why tests are failing (commitlint option in answers.linters)

### CODE REVIEW

**Review by pi0 — COMMENTED** [review]
I like the idea :+1:

**Comment by vinayakkulkarni** [comment]
We could even add Circle CI, Travis CI etc. in future!

**Review by Atinux — APPROVED** [review]

File: packages/cna-template/template/frameworks/github-actions/.github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+      <%_ } _%>
+
+      - name: Install dependencies 👨🏻‍💻
+        run: <%= pm === 'yarn' ? 'yarn' : 'npm' %> ci
```
Comment (clarkdo) [inline packages/cna-template/template/frameworks/github-actions/.github/workflows/ci.yml:56]: Can be `pmRun`.

**Review by clarkdo — APPROVED** [review]
Nice!

File: packages/create-nuxt-app/lib/saofile.js [hunk 2]
```diff
@@ -76,6 +76,18 @@ module.exports = {
       })
     }
 
+    const eslint = this.answers.linter.includes('eslint')
+    const prettier = this.answers.linter.includes('prettier')
+    const lintStaged = eslint && this.answers.linter.includes('lintStaged')
+    const stylelint = this.answers.linter.includes('stylelint')
+    if (this.answers.ci !== 'none' && (this.answers.test !== 'none' || eslint || prettier || lintStaged || stylelint)) {
```
Comment (clarkdo) [inline packages/create-nuxt-app/lib/saofile.js:83]: I think this can be sth like `this.answers.linter.length > 0`

Comment (vinayakkulkarni) [reply packages/create-nuxt-app/lib/saofile.js:83]: initially I did do that but the tests were failing, can you please try and reproduce it at your end?

I've added you as collaborator to https://github.com/vinayakkulkarni/create-nuxt-app

Comment (clarkdo) [reply packages/create-nuxt-app/lib/saofile.js:83]: NP, I’ll try it today

File: packages/cna-template/template/frameworks/github-actions/.github/workflows/ci.yml (hunk 1 above)
Comment (vinayakkulkarni) [reply packages/cna-template/template/frameworks/github-actions/.github/workflows/ci.yml]: - [x] done in c90f1e4

Comment (clarkdo) [reply packages/cna-template/template/frameworks/github-actions/.github/workflows/ci.yml]: Sorry that I may not explain clearly, pmRun is a wapper for both yarn and npm, the value is  `yarn` or `npm run`, so what we could do is:

```js
// From
<%= pm === 'yarn' ? 'yarn' : 'npm' %> ci
// To
<%= pmRun %> ci
```

Comment (vinayakkulkarni) [reply packages/cna-template/template/frameworks/github-actions/.github/workflows/ci.yml]: yes, sorry, I am half asleep due to sleep deprivation.. I've fixed it in e62774a

File: packages/create-nuxt-app/lib/saofile.js (hunk 2 above)
Comment (clarkdo) [reply packages/create-nuxt-app/lib/saofile.js]: @vinayakkulkarni Updated, can you please review ?

Comment (vinayakkulkarni) [reply packages/create-nuxt-app/lib/saofile.js]: LGTM

**Pull request merged**